src/plugins/files/server/blob_storage_service/adapters/es/es.ts

```diff
@@ -38,14 +38,21 @@ interface UploadOptions {
 }
 
 export class ElasticsearchBlobStorageClient implements BlobStorageClient {
-  private static defaultSemaphore: Semaphore;
+  private static defaultUploadSemaphore: Semaphore;
+  private static defaultDownloadSemaphore: Semaphore;
 
   /**
-   * Call this function once to globally set a concurrent upload limit for
+   * Call this function once to globally set the concurrent transfer (upload/download) limit for
    * all {@link ElasticsearchBlobStorageClient} instances.
    */
-  public static configureConcurrentUpload(capacity: number) {
-    this.defaultSemaphore = new Semaphore(capacity);
+  public static configureConcurrentTransfers(capacity: number | [number, number]) {
+    if (Array.isArray(capacity)) {
+      this.defaultUploadSemaphore = new Semaphore(capacity[0]);
+      this.defaultDownloadSemaphore = new Semaphore(capacity[1]);
+    } else {
+      this.defaultUploadSemaphore = new Semaphore(capacity);
+      this.defaultDownloadSemaphore = new Semaphore(capacity);
+    }
```

Comment:
I opted for this approach because it's ideally supposed to be called only once, and also enforces intentional defaults for upload and download capacity in one go. it's also worth pointing out semaphores with alternate capacities other than the provisioned default can be passed for each instance of `ElasticsearchBlobStorageClient` created.